### PR TITLE
Deprecate duplicate artifact download APIs

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Deprecate duplicate artifact download APIs.

The following APIs are removed:
```
MlflowClient.download_artifacts
```

## How is this patch tested?

Unit tests.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Deprecate duplicate artifact download API: `MlflowClient.download_artifacts`

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [x] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by WeichenXu123** [comment]
shall we add an argument “tracking_uri” for `mlflow.artifacts.download_artifacts` ? Otherwise we need to globally set tracking uri (by env variable or call `mlflow.set_tracking_uri`) before calling the API.

**Comment by dbczumar** [comment]
> shall we add an argument “tracking_uri” for `mlflow.artifacts.download_artifacts` ? Otherwise we need to globally set tracking uri (by env variable `MLFLOW_TRACKING_URI` or call `mlflow.set_tracking_uri`) before calling the API.

Ah, that's definitely a problem. Good catch! Let's keep these two APIs for now. Are there any other duplicates we can remove?

File: mlflow/artifacts/__init__.py [hunk 1]
```diff
@@ -52,7 +55,12 @@ def download_artifacts(
         return _download_artifact_from_uri(artifact_uri, output_path=dst_path)
 
     artifact_path = artifact_path if artifact_path is not None else ""
-    store = _get_store()
+
+    if tracking_uri is not None:
+        with _use_tracking_uri(tracking_uri):
+            store = _get_store()
+    else:
+        store = _get_store()
```
Comment (harupy) [inline mlflow/artifacts/__init__.py:63]:
```suggestion
    store = _get_store(tracking_uri)
```

Can we pass `tracking_uri` to `_get_store`?

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -1483,6 +1483,7 @@ def print_artifact_info(artifact):
         """
         return self._tracking_client.list_artifacts(run_id, path)
 
+    @deprecated("mlflow.artifacts.download_artifacts", "2.0")
```
Comment (harupy) [inline mlflow/tracking/client.py:1486]: The warning message looks incorrect. It should be `mlflow.tracking.client.MflowClient.download_artifacts`

image

**Review by harupy — APPROVED** [review]
Left https://github.com/mlflow/mlflow/pull/6537#discussion_r952130169, otherwise LGTM!

File: mlflow/tracking/client.py (hunk 2 above)
Comment (harupy) [reply mlflow/tracking/client.py:1486]: We can address this issue in a separate PR.

Comment (WeichenXu123) [reply mlflow/tracking/client.py:1486]: I will file a follow-up PR to fix this.

Comment (harupy) [reply mlflow/tracking/client.py:1486]: You don't need to because I already did in #6552.

**Pull request merged**